tests/parsers/test_header.py

```diff
@@ -121,6 +121,19 @@ def test_get_summary(self):
                 self.assertEqual(header_parser.get_summary(), expected_summary)
                 self.assertEqual(header_parser.get_parameters(), expected_parameters)
 
+    def test_start_patterns(self):
+        """Check the header parser properly guards later patterns. This is
+        important so that when parsing multiple logs, the header parser does
+        not interrupt.
+
+        The below presolve line can be caught by the model name/size parser
+        in the header, but it should only be picked up if the HeaderParser
+        has seen a proper log start line."""
+
+        parser = HeaderParser()
+        parse_lines(parser, ["Presolved: 390 rows, 316 columns, 1803 nonzeros"])
```

Comment:
I have seen this to happen in the dev log when we have node presolve output too. If I am not overlooking, in a typical log, the header parser should not interrupt because we should have been in the presolve parser when reaching this line. That being said, thank you for the change. It is definitely safer to guard against some patterns.